src/http/headers.cr

```diff
@@ -55,37 +56,72 @@ struct HTTP::Headers
     @hash = Hash(Key, String | Array(String)).new
   end
 
-  def []=(key, value : String)
+  # Sets the value of key to the given value.
+  # ArgumentError will be raised if header includes invalid characters.
+  #
+  # ```
+  # headers = HTTP::Headers{"host" => "crystal-lang.or"}
+  # headers["host"] = "crystal-lang.org"
+  # headers["accept"] = "text/html"
+  # headers #=> HTTP::Headers{"host" => "crystal-lang.org", "accept" => "http/html"}
+  # ```
+  def []=(key : String, value : String) : String
     check_invalid_header_content(value)
 
     @hash[wrap(key)] = value
   end
 
-  def []=(key, value : Array(String))
+  # Sets the value of key to the given value.
```

Comment:
Perhaps "Given values"? Maybe show how the header key serializes with multiple values by showing `headers["host"]`?